README_GITHUB_ACTIONS.md

```diff
@@ -0,0 +1,38 @@
+# GitHub Actions Notes for Maintainers
+
+rocket-chip makes use of [GitHub Actions](https://github.com/features/actions)
+for running continuous integration (CI) tests. For more information on GitHub
+Actions, please see their [documentation](https://help.github.com/en/actions).
+
+
+## How It Is Configured
+
+The main configuration file describing the continuous integration workflow is
+located at
+[.github/workflows/continuous-integration.yml](.github/workflows/continuous-integration.yml).
+This sets up a few jobs for prebuilding and caching a few tools, such as the
+riscv-tools and Verilator, as well as a large matrix job that runs the suite of
+tests in parallel.
+
+The actual Makefile targets run as part of the test suites defined in
+[regression/run-test-bucket](regression/run-test-bucket).
+
+For information on the Workflow file syntax, please see the
+[API reference](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions).
+
+
+## Caching
+
+Caching is handled by the official GitHub Action [actions/cache](https://github.com/marketplace/actions/cache).
+Caches are kept around for one week, and caches are searched in order branch,
+base branch (e.g. PR target), and default branch (e.g. master).
+
+[There is no way to delete a cache](https://github.com/actions/cache/issues/294),
+so the recommended approach for invalidating a cache is to modify the cache key.
+In the rocket-chip GitHub Action workflow configuration file, the cache keys
+each end in a number. These numbers can be incremented to "invalidate" the
```

Comment:
I am trying to understand exactly what you have to change. Are you saying you'd change `-v1` to `-v2` e.g. here: https://github.com/chipsalliance/rocket-chip/pull/2459/files#diff-17b09a7990ba40f6a89628ab9402f1f5R150. If so, perhaps you could change this to say "each end in a version, e.g.` -v1`. These version numbers can be incremented..." If you mean something else it might be nice to clarify since this seems like something users may need to do even if they aren't that familiar with the Github Actions API

Reply:
Thanks for the feedback. I had a feeling that what I wrote wouldn't be clear. I updated the descriptions to give an example of the version number and to include step-by-step instructions for invaliding the cache: 1e2fe619ea35f8bc6a417372049878bd4b1b5679

Can you take a look and let me know if that's any better?

Reply:
yep, that's very clear. Thanks!

Reply:
one final thing... what's an example "However, there are situations where
you may need to manually invalidate the caches."

Like, is it a situation where you need to invalidate your cache because you were doing something, then rebased on master which had updated the riscv-tools.hash and therefore you want to grab the master cache instead of your PR cache? But that seems like not a scenario since when you rebsae on master your riscv-tools.hash will get updated...

I guess maybe the answer is "there may be uncommon scenarios"

Reply:
No, the scenario I was thinking of was the scenario I hit over and over again while developing this PR, which was that any changes to the actual configuration of the riscv-tools compile steps (Makefile rules) or dependent packages will _not_ get picked up.

I suppose that I could include the Makefile rules in the hash computation, but I was afraid that would lead to too many false invalidations. I am, however, now thinking that putting the `apt-get install` stuff into a standalone script is not a bad idea, since that would allow us to more properly invalidate the cache if someone changes the dependencies.

I'll probably work on that, but I might now have much time to work on this PR for a couple days.